test/cancelledTradeProxy.test.ts

```diff
@@ -0,0 +1,74 @@
+process.env.SYSTEM_EVENT_BUS_NAME = 'mytesteventbus';
+import {mockClient} from 'aws-sdk-client-mock';
+import {handler} from '../assets/lambda/cancelledTradeProxy';
+import {createEvent} from './fakeApiGatewayEvent';
+import {EventBridgeClient, PutEventsCommand} from '@aws-sdk/client-eventbridge';
+const eventBridgeMock = mockClient(EventBridgeClient);
+
+beforeEach(() => {
+  eventBridgeMock.on(PutEventsCommand).resolves({
+    FailedEntryCount: 0,
+  });
+});
+
+afterEach(() => {
+  eventBridgeMock.reset();
+});
+
+test('Should raise event for agency cancel', async () => {
+  const body = {
+    order_id: '111',
+    mgi_version: '12',
+    mgi_last_moditifed_time: '2021-8-16',
+    mgi_cancellation_reason: 'rebook',
+    trading_model: 'agency',
+  };
+
```

Comment:
let be sure what that execution environment has proper state before making a call.
```suggestion
expect(eventBridgeMock.calls()).toHaveLength(0);
```

Reply:
done